### DISCUSSION TITLE
[Question][Hw1] Why there is no input function in domain.ml?

### DISCUSSION THREAD

**Discussion by jsjclink** [prosyslab-classroom/cs424-program-reasoning#386] [Q&A]
Hi, I have a question about homework 1.

Why there is no definition of input function in domain.ml?

In the module VALUE, there is definition of function "print" (`val print : t -> unit`).
But I can't find the definition of function "input".

Can I just use `input_line ` in the semantics.ml(in the `Llvm.Opcode.Call when Llvmutils.is_input instr`)?

**Comment by Hhro** [comment]
Hi,

To get input from the user, you can use any function like `read_line`, `read_int`, and e.t.c.

By the way, I don't know what you mean by `input_line`. Is it a typo of `read_line`?

Thanks.

**Comment by jsjclink** [comment]
Thank you!
input_line is also a function that reads input.  
I'll use read_line
